amgut/static/js/interactive.js

```diff
@@ -0,0 +1,292 @@
+$(document).ready(function() {
+  //Patching in Mth.log2
+  Math.log2 = Math.log2 || function(x) {
+                return Math.log(x) / Math.LN2;
+            };
+  //Patching in findIndex for Internet Explorer
+  if (!Array.prototype.findIndex) {
+    Array.prototype.findIndex = function(predicate) {
+      if (this === null) {
+        throw new TypeError('Array.prototype.findIndex called on null or undefined');
+      }
+      if (typeof predicate !== 'function') {
+        throw new TypeError('predicate must be a function');
+      }
+      var list = Object(this);
+      var length = list.length >>> 0;
+      var thisArg = arguments[1];
+      var value;
+
+      for (var i = 0; i < length; i++) {
+        value = list[i];
+        if (predicate.call(thisArg, value, i, list)) {
+          return i;
+        }
+      }
+      return -1;
+    };
+  }
+});
+
+function group(list, prop) {
+  var grouped = {};
+  for(var i=0;i<list.length;i++) {
+    var key = list[i][prop];
+    grouped[key] = grouped[key] || [];
+    grouped[key].push(list[i]);
+  }
+  return grouped;
+}
+
+function getAvg(list) {
+  return list.reduce(function (x, y) {
+    return x + y;
+  }) / list.length;
+}
+
+function sumArrays(ar1, ar2) {
+  var sum = [];
+  if (ar1.length == ar2.length) {
+      for (var i=0; i<ar1.length;i++) {
+          sum.push(ar1[i] + ar2[i]);
+      }
+  }
+  else {
+    throw "Arrays unequal length: " + ar1.length + "  " + ar2.length;
+  }
+  return sum;
+}
+
+function findOtu(list, otu, level) {
+  var lvl = level || 'label';
+  function inner(element, index, array) {
+    if(element[lvl] == this) { return true; }
+    return false;
+  }
+  return list.findIndex(inner, otu);
+}
+
+function filterSites(list, sites) {
+
+}
+
+function collapse(dataset, level, max, prev_level, focus, sites) {
+  var phylum_colors = {
```

Comment:
How painful would it be for these to not be hard coded? I ask because the display is essentially specific to human associated